packages/menu-bar/src/vaadin-menu-bar-mixin.js

```diff
@@ -749,4 +758,51 @@ export const MenuBarMixin = (superClass) =>
         this._subMenu.close();
       }
     }
+
+    /** @private */
+    _requestClose() {
+      this._preventClose = false;
+      // Wait if something has to prevent the close event
+      setTimeout(() => {
+        if (!this._preventClose) {
+          this._close(false);
+        }
+      }, 300);
+    }
+
+    /** @private */
+    _addHoverListener(subMenu) {
+      const menuOverlay = subMenu.$.overlay.$.overlay;
+
+      menuOverlay.addEventListener(
+        'mouseleave',
+        () => {
+          this._requestClose();
+        },
+        { once: true },
+      );
+      // When hovering between sub menus the subMenu will close if we don't prevent it
+      menuOverlay.addEventListener(
+        'mouseenter',
+        () => {
+          this._preventClose = true;
+        },
+        { once: true },
+      );
```

Comment:
Same as https://github.com/vaadin/web-components/pull/1979#discussion_r981045257

The event listeners keep adding up:
![Screenshot 2023-01-10 at 18 04 19](https://user-images.githubusercontent.com/1222264/211601922-1a6b7674-d521-40b9-9270-2354dafca9b1.png)

Reply:
> I made some changes to fix the problems that you`ve found.

Hi. It still seems to add more listeners on every submenu open:

https://user-images.githubusercontent.com/1222264/211859848-65ca2c03-272e-4e46-adaa-f1bfa9331be5.mp4

Could be easier to always add the hover listeners to each submenu (a single time), and then check the `this.openOnHover` state inside the listener (while unlikely, it's possible to change the value of `this.openOnHover` dynamically).